engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/SnapshotServiceImpl.java

```diff
@@ -232,27 +242,27 @@ protected Void createSnapshotAsyncCallback(AsyncCallbackDispatcher<SnapshotServi
     public SnapshotResult takeSnapshot(SnapshotInfo snap) {
         SnapshotObject snapshot = (SnapshotObject)snap;
 
-        SnapshotObject snapshotOnPrimary = null;
+        SnapshotObject snapshotOnStorage = null;
```

Comment:
```suggestion
        SnapshotObject snapshotOnPrimaryStorage = null;
```

if this object is for the snapshot on primary storage